Rubberduck.Core/UI/Command/SyncCodeExplorerCommand.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System.Linq;
+using System.Windows;
+using NLog;
+using Rubberduck.Navigation.CodeExplorer;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.UI.Command
+{
+    public interface ICodeExplorerSyncProvider
+    {
+        SyncCodeExplorerCommand GetSyncCommand(CodeExplorerViewModel explorer);
+    }
+
+    public class CodeExplorerSyncProvider : ICodeExplorerSyncProvider
+    {
+        private readonly IVBE _vbe;
+        private readonly RubberduckParserState _state;
+
+        public CodeExplorerSyncProvider(IVBE vbe, RubberduckParserState state)
+        {
+            _vbe = vbe;
+            _state = state;
+        }
+
+        public SyncCodeExplorerCommand GetSyncCommand(CodeExplorerViewModel explorer)
+        {
+            return new SyncCodeExplorerCommand(_vbe, _state, explorer);
+        }
+    }
+
+    public class SyncCodeExplorerCommand : CommandBase
+    {
+        private readonly IVBE _vbe;
+        private readonly RubberduckParserState _state;
+        private readonly CodeExplorerViewModel _explorer;
+
+        public SyncCodeExplorerCommand(IVBE vbe, RubberduckParserState state, CodeExplorerViewModel explorer) : base(LogManager.GetCurrentClassLogger())
+        {
+            _vbe = vbe;
+            _state = state;
+            _explorer = explorer;
+        }
+
+        protected override bool EvaluateCanExecute(object parameter)
+        {
+            if (_state.Status != ParserState.Ready || 
+                _explorer.IsBusy || 
+                //_explorer.TreeViewVisibility != Visibility.Visible ||
```

Comment:
Delete the comment.

Reply:
Done.  I'll be shocked if that was the only one I missed.